packages/flutter/lib/src/rendering/table.dart

```diff
@@ -703,7 +705,9 @@ class RenderTable extends RenderBox {
     if (_rowDecorationPainters != null) {
       for (BoxPainter painter in _rowDecorationPainters)
         painter?.dispose();
-      _rowDecorationPainters = null;
+      // We need to reset the _rowDecorationPainters by using the setter to ensure
+      // autonomy.
+      rowDecorations = null;
```

Comment:
It seems odd that we change the configuration of the RenderObject. Aren't the rowDecorations user-provided?